src/Interpreters/HashJoin/HashJoin.cpp

```diff
@@ -1696,18 +1696,42 @@ void HashJoin::tryRerangeRightTableDataImpl(Map & map [[maybe_unused]])
 
             auto & columns = columns_list.back().columns;
             size_t start_row = columns.at(0)->size();
-            for (; it.ok(); ++it)
+            size_t new_rows = 0;
```

Comment:
Could you pls elaborate on what the issue was exactly, and how this change will solve it?